x/staking/proto/cosmos/staking/v1beta1/query.proto

```diff
@@ -139,10 +139,17 @@ message QueryValidatorsRequest {
   cosmos.base.query.v1beta1.PageRequest pagination = 2;
 }
 
+message ValidatorInfo {
+  option (cosmos_proto.message_added_in) = "cosmos-sdk 0.50";
+  Validator validator = 1 [(gogoproto.nullable) = false, (amino.dont_omitempty) = true];
+  // consensus_address is the consensus address of the validator.
+  string consensus_address = 2 [(cosmos_proto.field_added_in) = "cosmos-sdk 0.50"];
+}
+
 // QueryValidatorsResponse is response type for the Query/Validators RPC method
 message QueryValidatorsResponse {
-  // validators contains all the queried validators.
-  repeated Validator validators = 1 [(gogoproto.nullable) = false, (amino.dont_omitempty) = true];
+  // validators contains all the queried validators with additional info.
+  repeated ValidatorInfo validators = 1 [(gogoproto.nullable) = false, (amino.dont_omitempty) = true];
```

Comment:
This breaks backwards compatibility (older client, newer server) in a probably high throughput query, are we really sure we want this?

Reply:
Are you suggesting something like this?
```
message QueryValidatorsResponse {
  repeated Validator validators = 1 [(gogoproto.nullable) = false, (amino.dont_omitempty) = true];

  repeated ValidatorInfo validator_info = 2 [(gogoproto.nullable) = false, (amino.dont_omitempty) = true];

  cosmos.base.query.v1beta1.PageResponse pagination = 3;
}

message ValidatorInfo {
  option (cosmos_proto.message_added_in) = "cosmos-sdk 0.52";
  Validator validator = 1 [(gogoproto.nullable) = false, (amino.dont_omitempty) = true];
  // consensus_address is the consensus address of the validator.
  string consensus_address = 2 [(cosmos_proto.field_added_in) = "cosmos-sdk 0.52"];
}
```

The problem that I see here is that we will be repeating all the validator information, and if we only include validatorInfo without the validator details, how do we associate the consensus key with the validator?

Regarding the performance concerns, maybe we could add a configuration option to enable or disable this?

@tac0turtle @kocubinski

Reply:
I'm not concerned with performance, higher throughput means on a breaking proto change means more client breaks.  Let's not break proto specs.  Suggesting

```proto
message QueryValidatorsResponse {
  repeated Validator validators = 1 [(gogoproto.nullable) = false, (amino.dont_omitempty) = true];

  repeated ValidatorInfo validator_info = 2 [(gogoproto.nullable) = false, (amino.dont_omitempty) = true];

  cosmos.base.query.v1beta1.PageResponse pagination = 3;
}

message ValidatorInfo {
  option (cosmos_proto.message_added_in) = "cosmos-sdk 0.52";
  // consensus_address is the consensus address of the validator.
  string consensus_address = 1 [(cosmos_proto.field_added_in) = "cosmos-sdk 0.52"];
}
```

Reader and writers correlate `validator_info` with `validators` through the use of an index.  Patterns like this are common for APIs which must not introduce breaking changes!

Reply:
done @kocubinski